src/uct/ib/base/ib_md.c

```diff
@@ -1239,8 +1239,7 @@ static ucs_status_t uct_ib_query_md_resources(uct_md_resource_desc_t **resources
     device_list = ibv_get_device_list(&num_devices);
     if (device_list == NULL) {
         ucs_debug("Failed to get IB device list, assuming no devices are present");
-        status = UCS_ERR_NO_DEVICE;
-        goto out;
+        num_devices = 0;
     }
 
     resources = ucs_calloc(num_devices, sizeof(*resources), "ib resources");
```

Comment:
can calloc return `NULL` now? If yes, the func will fail with `UCS_ERR_NO_MEMORY`